experimental/LieAlgebras/src/simple_lie_algebra.jl

```diff
@@ -0,0 +1,263 @@
+@attributes mutable struct SimpleLieAlgebra{C<:RingElement} <: LieAlgebra{C}
+#Construct a simple Lie algebra over a given ring with a given root system
+  mat_space::MatSpace
+  base_ring::Ring
+  root_system::RootSystem
+  dim::Int
+  s::Vector{Symbol}
+  root_type::Tuple{String,Int64}
+
+  function SimpleLieAlgebra{C}(R::Ring, S::String, cached::Bool) where {C<:RingElement}
+    RS = RootSystem(S)
+    di = number_of_roots(RS) + length(RS.simple_roots)
+    M = MatrixSpace(R, 1, di)
+    s = [Symbol("e_$i") for i in 1:di]
+    st = RS.root_system_type
+    return get_cached!(SimpleLieAlgebraDict, (R, RS, M), cached) do
+    new{C}(M, R, RS, di, s, st)
+    end::SimpleLieAlgebra{C}
+  end
+end
+
+const SimpleLieAlgebraDict = CacheDictType{
+  Tuple{Ring,RootSystem,MatSpace},SimpleLieAlgebra
+}()
+
+mutable struct SimpleLieAlgebraElem{C<:RingElement} <: LieAlgebraElem{C}
+  parent::SimpleLieAlgebra{C}
+  mat::MatElem{C}
+  function SimpleLieAlgebraElem{C}(
+    L::SimpleLieAlgebra{C},
+    M::MatElem{C}) where {C<:RingElement}
+    return new{C}(L, M)
+  end
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+parent_type(::Type{SimpleLieAlgebraElem{C}}) where {C<:RingElement} = SimpleLieAlgebra{C}
+
+elem_type(::Type{SimpleLieAlgebra{C}}) where {C<:RingElement} = SimpleLieAlgebraElem{C}
+
+parent(x::SimpleLieAlgebraElem{C}) where {C<:RingElement} = x.parent
+
+base_ring(x::SimpleLieAlgebraElem{C}) where {C<:RingElement} = x.parent.base_ring
+
+base_ring(L::SimpleLieAlgebra{C}) where {C<:RingElement} = L.base_ring::parent_type(C)
+
+dim(L::SimpleLieAlgebra{C}) where {C<:RingElement} = L.dim
+
+root_system(L::SimpleLieAlgebra{C}) where {C<:RingElement} = L.root_system
+
+root_type(L::SimpleLieAlgebra{C}) where {C<:RingElement} = L.root_type
+
+matrix(f::SimpleLieAlgebraElem) = f.mat
+
+@doc raw"""
+    characteristic(L::SimpleLieAlgebra{T}) -> Int64
+
+Return the characteristic of the base ring of the Lie algebra `L`.
+"""
+characteristic(L::SimpleLieAlgebra{T}) where T <: RingElement = characteristic(base_ring(L))
+
+# linear indexing 
+getindex(f::SimpleLieAlgebraElem, r::Int, c::Int) = f.mat[r, c]
+
+Base.@propagate_inbounds function Base.setindex!(f::SimpleLieAlgebraElem{T}, d, r) where {T <: RingElem}
+  matrix(f)[1, r] = d
+  return f
+end
```

Comment:
I'm going to be working a lot with root vectors where the entries in the matrix-part of a Lie algebra element correspond to a coefficient of a root vector.  It would be helpful to be able to just change certain coefficients very quickly. It can also be done in a different way, I have commented this part out for now.